bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/handler/HaywardBackyardHandler.java

```diff
@@ -0,0 +1,88 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.haywardomnilogic.internal.HaywardBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The Filter Handler
+ *
+ * @author Matt Myers - Initial Contribution
+ */
+@NonNullByDefault
+public class HaywardBackyardHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(HaywardBackyardHandler.class);
+
+    public HaywardBackyardHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!(command instanceof RefreshType)) {
+            String systemID = getThing().getProperties().get(HaywardBindingConstants.PROPERTY_SYSTEM_ID);
+            String poolID = getThing().getProperties().get(HaywardBindingConstants.PROPERTY_BOWID);
+            HaywardBridgeHandler haywardOmniLogixBridgeHandler = getHaywardOmniLogixBridgeHandler();
+            if (haywardOmniLogixBridgeHandler != null) {
+                haywardOmniLogixBridgeHandler.haywardCommand(channelUID, command, systemID, poolID);
+            }
+        }
+    }
+
+    public void updateBackyardData(String systemID, String channelID, String data) {
+        Channel chan = getThing().getChannel(channelID);
+        if (chan != null) {
+            String acceptedItemType = chan.getAcceptedItemType();
+            if (acceptedItemType != null) {
+                State state = toState(acceptedItemType, data);
+                updateState(chan.getUID(), state);
+                logger.trace("Updated Hayward Backyard {} {} to: {}", systemID, channelID, data);
+            }
+        }
+    }
+
+    private State toState(String type, String value) throws NumberFormatException {
+        if ("Number".equals(type)) {
+            return new DecimalType(value);
+        } else if ("Switch".equals(type)) {
+            return Integer.parseInt(value) > 0 ? OnOffType.ON : OnOffType.OFF;
+        } else {
+            return StringType.valueOf(value);
+        }
+    }
+
+    @SuppressWarnings("null")
+    private @Nullable HaywardBridgeHandler getHaywardOmniLogixBridgeHandler() {
+        return (HaywardBridgeHandler) getBridge().getHandler();
```

Comment:
This code was moved to haywardThingHandler.  Let me know if this is what you are looking for...  I'm under the impression we want to return a null if the bridgehandler is not found, but avoid the null point exception.   Right?

`    private HaywardBridgeHandler getHaywardOmniLogixBridgeHandler() {
        Bridge bridge = getBridge();
        if (bridge == null || bridge.getHandler() == null) {
            return null;
        } else {
            return (HaywardBridgeHandler) getBridge().getHandler();
        }
    }`

Reply:
If you do this approach, you need to use the local variable `bridge` in the return statement to avoid a compiler warning, as `getBridge()` could be set to null from another thread in the meantime.

Another approach would be to throw an exception if something is null. It's up to you what's easier to implement.